model/adb_auth.go

```diff
@@ -99,7 +124,64 @@ func getUsersJSON(db *sqlx.DB, options GetUserOptions) ([]UserJSON, error) {
 }
 
 func GetUsers(db *sqlx.DB, options GetUserOptions) ([]ADBUser, error) {
-	return getUsers(db, options)
+	users, err := getUsers(db, options)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if len(users) == 0 {
+		return users, nil
+	}
+
+	usersRolesOptions := GetUsersRolesOptions{
+		Users: users,
+	}
+
+	usersRoles, err := getUsersRoles(db, usersRolesOptions)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if len(usersRoles) == 0 {
+		return users, nil
+	}
+
+	/*
+		userIDToIndex := map[int]int{}
+		for i, user := range users {
+			userIDToIndex[user.ID] = i
+		}
+	  fmt.Println(userIDToIndex);
+
+		for _, r := range usersRoles {
+	    fmt.Println(r.UserID)
+			a := userIDToIndex[r.UserID]
+	    fmt.Println(a)
+	    fmt.Println(r)
+			users[a].Roles = append(users[a].Roles, r)
+		}
+	*/
+
+	// NOTE: This is a O(N2) operation at best.
+	// Encountered weird behavior with using userIDToIndex
+	// to determine the roles for each user.
+	// Example:
+	// When userIDToIndex = [7:0]
+	// userIDToIndex[6] would evaluate to 0. However, it should
+	// NOT find any index where UserID is 6.
```

Comment:
Map indexing on an absent key returns the zero value. You can distinguish this default value by using the two-value map index assignment form:

    v, ok := myMap[key]

Here `ok` will be true if `key` is present in the map.

Reply:
👍